orm/versioning_bucket.go

```diff
@@ -0,0 +1,84 @@
+package orm
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+)
+
+const latestVersionIndexName = "latest"
+
+type VersioningBucket struct {
```

Comment:
Okay, interesting idea here. And good use to capture all code. 

I see the complexities of maintaining full objects as multiple versions. Powerful, but maybe overly complex. I am happy to discuss a simpler alternative today.